src/gentropy/dataset/l2g_prediction.py

```diff
@@ -126,3 +126,54 @@ def to_disease_target_evidence(
                 "studyLocusId",
             )
         )
+
+    def add_locus_to_gene_features(
+        self: L2GPrediction, feature_matrix: L2GFeatureMatrix
+    ) -> L2GPrediction:
+        """Add features to the L2G predictions.
+
+        Args:
+            feature_matrix (L2GFeatureMatrix): Feature matrix dataset
+
+        Returns:
+            L2GPrediction: L2G predictions with additional features
+        """
+        # Testing if `locusToGeneFeatures` column already exists:
+        if "locusToGeneFeatures" in self.df.columns:
+            self.df = self.df.drop("locusToGeneFeatures")
+
+        # Columns identifying a studyLocus/gene pair
+        prediction_id_columns = ["studyLocusId", "geneId"]
+
+        # L2G matrix columns to build the map:
+        columns_to_map = [
+            column
+            for column in feature_matrix._df.columns
+            if column not in prediction_id_columns
+        ]
+
+        # Aggregating all features into a single map column:
+        aggregated_features = (
+            feature_matrix._df.withColumn(
+                "locusToGeneFeatures",
+                f.create_map(
+                    *sum(
```

Comment:
😨  I would not think of the expression like `sum(List[Tuple[Col, Col]], Tuple[]). Nice, although i it seems to be the way to flatten. https://stackoverflow.com/questions/10632839/transform-list-of-tuples-into-a-flat-list-or-a-matrix